hoomd/md/ForceComposite.cc

```diff
@@ -605,8 +606,8 @@ void ForceComposite::createRigidBodies()
         unsigned int central_particle_tag = 0;
         for (unsigned int particle_tag = 0; particle_tag < n_without_constituent; ++particle_tag)
             {
-            assert(snap.type[i] < ntypes);
-            assert(snap.body[i] == NO_BODY);
+            assert(snap.type[particle_tag] < ntypes);
```

Comment:
```suggestion
            assert(snap.type[particle_tag] < m_pdata->getNTypes());
```
Call it directly here to avoid the warning.